src/model.js

```diff
@@ -1,62 +1,99 @@
+export const Variable = (name) => node(Variable)({ name })
+export const Literal = (value) => node(Literal)({ value })
+export const List = (...values) => node(List)({ values })
+export const Closure = (...parameters) => (...sentences) => node(Closure)({ parameters, sentences: Block(...sentences) })
 
-export const Send = (target, key) => (...parameters) => Node(Send)({ target, key, parameters })
-export const Super = (...parameters) => Node(Super)({ parameters })
+export const Send = (target, key) => (...parameters) => node(Send)({ target, key, parameters })
+export const Super = (...parameters) => node(Super)({ parameters })
 
-export const New = (target) => (...parameters) => Node(New)({ target, parameters })
+export const New = (target) => (...parameters) => node(New)({ target, parameters })
 
-export const If = (condition) => (...thenSentences) => (...elseSentences) => Node(If)({ condition, thenSentences: Block(...thenSentences), elseSentences: Block(...elseSentences) })
-export const Throw = (exception) => Node(Throw)({ exception })
-export const Try = (...sentences) => (...catches) => (...always) => Node(Try)({ sentences: Block(...sentences), catches, always: Block(...always) })
-export const Catch = (variable, errorType) => (...handler) => Node(Catch)({ variable, errorType, handler: Block(...handler) })
+export const If = (condition) => (...thenSentences) => (...elseSentences) => node(If)({ condition, thenSentences: Block(...thenSentences), elseSentences: Block(...elseSentences) })
+export const Throw = (exception) => node(Throw)({ exception })
+export const Try = (...sentences) => (...catches) => (...always) => node(Try)({ sentences: Block(...sentences), catches, always: Block(...always) })
+export const Catch = (variable, errorType) => (...handler) => node(Catch)({ variable, errorType, handler: Block(...handler) })
 
-//-------------------------------------------------------------------------------------------------------------------------------
-// COMMONS
-//-------------------------------------------------------------------------------------------------------------------------------
+//===============================================================================================================================
+// CATEGORIES
+//===============================================================================================================================
+
+export const TopLevel = [Import, Package, Class, Mixin, Singleton, Program, Test]
+export const Member = [Field, Method, Constructor]
+export const Sentence = [VariableDeclaration, Return, Assignment]
+export const Expression = [Variable, Literal, List, Closure, Send, Super, New, If, Throw, Try]
+export const Node = [File, ...TopLevel, ...Member, ...Sentence, ...Expression, Catch, Parameter, Block]
 
-export const Parameter = (name, varArg = false) => Node(Parameter)({ name, varArg })
+Node.forEach(builder => { builder.toString = () => builder.name })
```

Comment:
I didn't get where you use categories later, since they are not part of the "name" or toString() of the nodes themselves, then once you have a Node, you cannot check if it is a TopLevel, for example ?

Or maybe doing `TopLevel.includes(node.type)` ? or something like that ?

oohh.. or using `node.is(TopLevel)` right ?